### PR TITLE
Add Frequently Asked Questions to docs

### PR DESCRIPTION
**Pull request by alubbock**
I've added some initial FAQs to the documentation, which we can build off of over time.

Based off of PR #329.

### CODE REVIEW

File: doc/faq.rst [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+`Gitter channel <https://gitter.im/pysb/pysb>`_.
+For more general Python related questions, we recommend `Stack
+Overflow <https://www.stackoverflow.com>`_.
+
+General
+=======
+
+* Do you support Python 3?
+
+    **Yes, with one caveat, which will be fixed in the next release**. PySB
```
Comment (jmuhlich) [inline doc/faq.rst]: Now that #309 has been merged, this whole answer can basically be replaced with "YES!"

Comment (alubbock) [reply doc/faq.rst]: Updated FAQ on Python 3 support. I left a note in place about the previous reversible rule syntax and that it needs to be replaced for Python 3 support.

File: doc/faq.rst [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+For more general Python related questions, we recommend `Stack
+Overflow <https://www.stackoverflow.com>`_.
+
+General
+=======
+
+* Do you support Python 3?
+
+    **Yes**. The current release of PySB supports Python 3.6 and
+    Python 2.7. Other Python 3.x releases but are not explicitly tested.
```
Comment (jmuhlich) [inline doc/faq.rst]: I think there's an extraneous "but" in this sentence.

File: doc/faq.rst [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+    value will come from the first condition which evaluates to True. Thus,
+    for the Expression to always have a value, the last condition should
+    default to True.
+
+Simulation
+==========
+
+* How can I speed up my `ScipyOdeSimulator` simulation?
+
+    **Check the weave library is installed.** `weave` is a Python library
```
Comment (jmuhlich) [inline doc/faq.rst]: Since #320 is imminent, maybe we should suggest installing cython instead?

Comment (jmuhlich) [reply doc/faq.rst]: Nevermind, as you said in the other PR, weave still seems faster. We'll resolve this later.

File: doc/faq.rst (hunk 2 above)
Comment (alubbock) [reply doc/faq.rst]: Fixed.

**Pull request merged**